src/platform/datapath_raw_socket.c

```diff
@@ -475,10 +475,17 @@ CxPlatResolveRoute(
 
 Done:
     if (Status != ERROR_IO_PENDING && Status != ERROR_SUCCESS) {
+        //
+        // Failed to resolve route. Queue route resolution completion with failure and
```

Comment:
This code here doesn't officially know that Callback will "queue" anything, so we shouldn't say anything about the queueing here. This is especially confusing because above in the ERROR_IO_PENDING path we really do queue something here.

Reply:
Hmm. I can move `Route->State = RouteResolving;` out of the ERROR_IO_PENDING path so we do it regardless. I think it makes sense because as long as route completion has not been called, we are still resolving the route.

Reply:
and remove the comment

Reply:
Good idea.